client/src/pages/PetDetail/index.js

```diff
@@ -36,21 +41,65 @@ class PetDetail extends Component {
       this.props.petsAddress[this.props.match.params.address]
     );
     this.setState({ petInstance: PetInstance });
-    await this.getPetInfo();
     this.stage = new createjs.Stage('canvas');
-    this.setState({ action: PetAction.DEFAULT });
-    this.action(PetAction.DEFAULT);
+    this.getPetInfo();
   }
   async getPetInfo() {
     let PetInstance = this.state.petInstance;
-    let id = await PetInstance.methods.petId().call();
-    this.setState({ type: id });
-    let amount = await PetInstance.methods.providentFund().call();
-    this.setState({ providentFund: amount });
-    let time = await PetInstance.methods.growthTime().call();
-    this.setState({ growthTime: time });
+    let PetInfo = await PetInstance.methods.getInfomation().call();
+    this.setState({
+      providentFund: PetInfo[1],
+      growthTime: PetInfo[2],
+      targetFund: PetInfo[3],
+      duration: PetInfo[4]
+    });
+    this.getProgress();
+    this.getSize();
+    this.action();
+  }
+  getProgress() {
+    let progress = (this.state.growthTime / this.state.duration) * 100;
```

Comment:
một pha xử lý quá cồng kềnh
```suggestion
progress = progress < 50  ? 0 : 1;
this.setState({progress})
```

Reply:
con này là để các tam thức bậc 3 lồng nhau à a ?

Reply:
lồng đâu, vẫn chỉ là 1 phép so sánh thôi mà